modules/openapi-generator/src/main/resources/C-libcurl/apiClient.c.mustache

```diff
@@ -12,21 +12,29 @@ apiClient_t *apiClient_create() {
     apiClient->basePath = "{{{basePath}}}";
     apiClient->dataReceived = NULL;
     apiClient->response_code = 0;
-    #ifdef BASIC_AUTH
+    {{#hasAuthMethods}}
+    {{#authMethods}}
+    {{#isBasic}}
     apiClient->username = NULL;
     apiClient->password = NULL;
-    #endif // BASIC_AUTH
-    #ifdef OAUTH2
+    {{/isBasic}}
+    {{#isOAuth}}
     apiClient->accessToken = NULL;
-    #endif // OAUTH2
+    {{/isOAuth}}
+    {{#isApiKey}}
+    apiClient->apiKeys = NULL;
+    {{/isApiKey}}
+    {{/authMethods}}
+    {{/hasAuthMethods}}
+
     return apiClient;
 }
 
 void apiClient_free(apiClient_t *apiClient) {
     if(apiClient->dataReceived) {
         free(apiClient->dataReceived);
     }
-    free(apiClient);
+    //free(apiClient);
```

Comment:
@zhemant maybe simply remove this line instead of commenting it out?

Reply:
I modified the code in new commits, now in API I free the dataReceived and leave apiClient object as it is for the user to free. So If someone doesn't want to reuse apiClient, instead of calling free function for multiple parameters, only 1 function will do it.